### ISSUE TITLE
Simultaneous attacks

### ISSUE THREAD

**Issue by exectails** [aura-project/aura#7]
Two creatures have to be able to hit each other simultaneously. Additionally, through the same mechanic, one creature A must be guaranteed to hit another creature B if B comes at A with Smash, effectively countering the attack.

Initially this was thought to be very tricky to implement, but research has shown that a creature is in "attack mode" from the moment you first click on an enemy, until you go out of battle mode. During that entire time you will counter Smash and have a chance to hit simultaneously.

**Comment by DeleteIT** [comment]
You have a CHANCE to counter Smash, and hit simultaneously, which is "Instinctive Reaction", IIRC.

**Comment by exectails** [comment]
What I'm referring to is mindlessly running into someone who's coming at you with smash and hitting him.

**Comment by DeleteIT** [comment]
Oh, I see, you've actioned the actual attack, and you're moving towards the target, but they have Smash loaded, and is coming at you?

In that case, yep, that's correct.

**Comment by xKamuna** [comment]
Yeah, I've already done this.  I can totally separate it from the super big pull request, if you want.  I've done the same for Normal Attack vs Normal Attack, though the mechanics for that are inferred since there is no information to go by other than what I can assume based on playing.

**Comment by exectails** [comment]
@xKamuna Of course, I want to encourage you very much to submit your changes in digestible pieces^^

**Comment by Xcelled** [comment]
What about a multiaggro situation? Like player clicks A and starts running toward A. B charges with smash loaded and attacks the player. Does the player intercept the smash?

If not (or regardless) we might want to implement a target check, so we only intercept smashes from the mob we're currently targeting. It always bugged me how you could 180-intercept a smash...

**Comment by xKamuna** [comment]
No, I have checks for that in place already.  No worries, I thought of that beforehand.

**Comment by xKamuna** [comment]
No, I already have checks for that in place.  I've thought of that already.  In official, I'm pretty sure you have to be targetting the Smasher in order to intercept it.  I think I tested that when I was doing this.

**Comment by exectails** [comment]
>It always bugged me how you could 180-intercept a smash...

Well, what's bugging you isn't really the key factor here, but how officials do it^^" We can't just change key game-play.

**Comment by exectails** [comment]
@xKamuna Do simultaneous attacks work like Smash, in that, when you click someone, then walk, and the other player tries to hit you, while you're still in "attack mode", does the server decide who hits? Or is the the actually attacking player's hit guaranteed?

**Comment by Sairii** [comment]
I might be totally wrong here:

Isn't the actual hit of smash delayed? Even tho the animation shows you that you already shoved your sword into a mob, the hit and probably stun only takes place at about half of the animation?
Isn't that what makes a normal attack win?

**Comment by exectails** [comment]
@Sairii That might be the "in-game explanation", the "logical reason" why it would be like that, but that's not what's happening behind the scenes. It's just a delayed animation, the server already calculated everything at the time the animation starts.

**Comment by xKamuna** [comment]
Yeah, because nobody likes timers- -shot-
But yes, the server calculates the winner before the attacks happen.

**Comment by exectails** [comment]
@xKamuna Was that the answer to my question?^^

**Comment by exectails** [comment]
@xKamuna Any update?

**Comment by boilerplate5** [comment]
If two entities run at each other, they have a chance to hit each other, or one side will hit the other and combat plays out as such. Smash should always be countered by a normal attack (or the new Bash skill I've read). The animation for Smash plays out but won't actually do anything and the user will be blown back soon enough.

**Comment by exectails** [comment]
@xKamuna I need to know if you'll handle this withing the next 2.5 weeks or if I have to do it.

**Comment by xKamuna** [comment]
Gasp, I'm actually needed?!  Yeah, I'm sure I can tear out the simultaneous attacks from my old repo, update it with some new stuff I thought of, and send up a pull request within 2.5 weeks.  Leave it to me!

**Comment by exectails** [comment]
Alright, thanks =)
